src/snowflake/snowpark/dataframe.py

```diff
@@ -726,7 +726,15 @@ def to_local_iterator(
         )
 
     def __copy__(self) -> "DataFrame":
-        return DataFrame(self._session, copy.copy(self._select_statement or self._plan))
+        if self._select_statement:
+            new_plan = copy.copy(self._select_statement)
+            new_plan.column_states = self._select_statement.column_states
```

Comment:
I think these should be deep copies.

Reply:
We had a discussion offline and these changes ideally should be in SelectStament.__copy__ but that will be a risk change (with multiple existing references to SelectStatement.__copy__) to check in last minute. The plan is to make these changes here for now and move it to SelectStatement.__copy__ in the Feburary's release.